src/Build/Utilities/EngineFileUtilities.cs

```diff
@@ -191,10 +191,11 @@ internal static bool FilespecHasWildcards(string filespecEscaped)
             FileMatcher.SearchAction action = FileMatcher.SearchAction.None;
             string excludeFileSpec = string.Empty;
 
-            if (!FilespecHasWildcards(filespecEscaped) ||
-                FilespecMatchesLazyWildcard(filespecEscaped, forceEvaluateWildCards))
+            var noWildcards = !FilespecHasWildcards(filespecEscaped) || FilespecMatchesLazyWildcard(filespecEscaped, forceEvaluateWildCards);
+            var noExcludeSpecs = excludeSpecsEscaped == null || !excludeSpecsEscaped.Any();
+            // It is possible to return original string if no entries in Exclude set and no wildcard matches. 
+            if (noWildcards && noExcludeSpecs)
```

Comment:
super-nit: Please consider taking advantage of short-circuit evaluation of conditions wherever practical. In this case `excludeSpecsEscaped.Any()` doesn't have to run if the filespec has wildcards.